Example/Example Swift/ComponentAPICoordinator.swift

```diff
@@ -287,6 +296,13 @@ extension ComponentAPICoordinator {
                     self.documentRequests[index].isUploaded = true
                 case .failure(let error):
                     self.documentRequests[index].error = error
+                    
+                    if self.documentRequests.type != .image || !self.giniConfiguration.multipageEnabled {
+                        guard let visionError = error as? GiniVisionError else { return }
+                        self.showErrorInAnalysisScreen(with: visionError.message, for: documentRequest) {
+                            self.upload(documentRequest: documentRequest)
```

Comment:
So, the `action` closure here is escaping. The method is called on the weak `self` from the outer closure and the inner closure also uses the same `self`. Are you sure that doesn't create a retain cycle?

Reply:
No, it doesn't 👍